rfcs/79-add-virtual-link-support.md

```diff
@@ -0,0 +1,56 @@
+- Author: abyss-w <mad.hatter@foxmail.com>
+- Start Date: 2021-08-16
+- RFC PR: [beyondstorage/go-service-qingstor#79](https://github.com/beyondstorage/go-service-qingstor/pull/79)
+- Tracking Issue: [beyondstorage/go-service-qingstor#64](https://github.com/beyondstorage/go-service-qingstor/issues/64)
+
+# RFC-79: Add Virtual Link Support
+
+## Background
+
+As you can see from the [official documentation for qingstor](https://docs.qingcloud.com/qingstor/), it does not support symbolic links itself. However, qingstor supports [user-defined object metadata](https://docs.qingcloud.com/qingstor/api/common/metadata), and we can use this to simulate the implementation of symlink.
+
+## Proposal
+
+I propose to use user-defined object metadata to support virtual link.
+
+```go
+input := &service.PutObjectInput{
+    XQSMetadata: &map[string]string{
+        "x-qs-meta-bs-link-target": rt,
+    },
+}
+```
+
+- `PutObjectInput` is used to store the fields we need when calling `PutObjectWithContext` API to upload an object.
+- `XQSMetadata` is a map that stores user-defined metadata.
+  - `"x-qs-meta-bs-symlink"` is the name of user-defined metadata, the middle `bs` is used to avoid conflicts.

```

Comment:
We also need to update `x-qs-meta-bs-symlink` to `x-qs-meta-bs-link-target` here.

Reply:
Got it.